packages/docusaurus-plugin-content-docs/src/docs.ts

```diff
@@ -64,9 +66,26 @@ async function readLastUpdateData(
 
     if (fileLastUpdateData) {
       const {author, timestamp} = fileLastUpdateData;
+
       return {
-        lastUpdatedAt: showLastUpdateTime ? timestamp : undefined,
-        lastUpdatedBy: showLastUpdateAuthor ? author : undefined,
+        lastUpdatedAt: (() => {
+          if (showLastUpdateTime) {
+            if (lastUpdateFrontMatter?.date) {
+              return new Date(lastUpdateFrontMatter.date).getTime() / 1000;
+            }
+            return timestamp;
+          }
+          return undefined;
+        })(),
+        lastUpdatedBy: (() => {
+          if (showLastUpdateAuthor) {
+            if (lastUpdateFrontMatter?.author) {
+              return lastUpdateFrontMatter.author;
+            }
+            return author;
+          }
+          return undefined;
+        })(),
```

Comment:
```suggestion
        lastUpdatedBy: showLastUpdateAuthor ? lastUpdateFrontMatter?.author ?? author : undefined,
```

Ditto for the other

Reply:
Should I have the `frontMatterTimestamp` variable like below to avoid nesting the ternary operators? I moved everything out of the `if (fileLastUpdateData)` because I realized that would mean that front matter `last_update` wouldn't work without Git.
```typescript
const {author, timestamp} = fileLastUpdateData ?? {};
const frontMatterTimestamp = (lastUpdateFrontMatter && lastUpdateFrontMatter.date) 
  ? (new Date(lastUpdateFrontMatter.date).getTime() / 1000) : undefined;

return {
  lastUpdatedBy: showLastUpdateAuthor ? lastUpdateFrontMatter?.author ?? author : undefined,
  lastUpdatedAt: showLastUpdateTime ? frontMatterTimestamp ?? timestamp : undefined,
}
```